src/osd/PrimaryLogPG.cc

```diff
@@ -11231,6 +11231,9 @@ bool PrimaryLogPG::start_recovery_ops(
   if (state_test(PG_STATE_RECOVERING)) {
     state_clear(PG_STATE_RECOVERING);
     state_clear(PG_STATE_FORCED_RECOVERY);
+    if (get_osdmap()->get_pg_size(info.pgid.pgid) <= acting.size()) {
+      state_clear(PG_STATE_DEGRADED);
```

Comment:
FYI, In the future I'm hoping to base PG_STATE_DEGRADED on the num_objects_degraded count.